erpnext/templates/pages/help.py

```diff
@@ -5,22 +5,74 @@
 
 def get_context(context):
 	context.no_cache = 1
+	context.bg = 'background-color: #fafbfc; border-radius:0'
+	context.align_greeting = 'start'
+	context.align_search_box = '0'
 	settings = frappe.get_doc("Support Settings", "Support Settings")
 	s = settings
 
+	context.greeting_text = s.greeting_text if s.greeting_text else "We're here to help"
+
+	if s.greeting_text_and_search_bar_alignment == 'Center':
+		context.align_greeting = 'center'
+		context.align_search_box = '25%'
+	if s.greeting_text_and_search_bar_alignment == 'Right':
+		context.align_greeting = 'end'
+		context.align_search_box = '50%'
+	if s.background == 'Color' and s.select_color:
+		context.bg = 'background-color: ' + s.select_color + '; border-radius:0'
+	if s.background == 'Image' and s.add_image:
+		context.bg = 'background-image: url(' + s.add_image + '); background-repeat: no-repeat; border-radius:0'
+
+	# Support content
+	favorite_article_count = 0
+	portal_setting = frappe.get_single("Portal Settings")
+	context.favorite_article_list=[]
+	context.help_article_list=[]
+	context.category_list = frappe.get_all("Help Category", fields="name")
+	all_articles = [i[0] for i in frappe.db.sql("""SELECT route from `tabHelp Article`""")]
+	favorite_articles = get_favorite_articles()
+	for article in favorite_articles:
+		favorite_article_dict = {}
+		if favorite_article_count < 3:
+			if article[0] in all_articles:
+				favorite_article = frappe.get_all("Help Article", fields=["title", "content", "route", "category"], filters={"route": article[0]})
+				content = frappe.utils.strip_html(favorite_article[0].content)
+				if len(content) > 115:
+					content = content[:112] + '...'	
+				favorite_article_dict = {
+					'title': favorite_article[0].title,
+					'content': content,
+					'category': favorite_article[0].category,
+					'route': favorite_article[0].route,
+				}
+				context.favorite_article_list.append(favorite_article_dict)
+				favorite_article_count += 1			
```

Comment:
This entire section needs a cleanup. 
If you want the top ten Help Articles for instance, write a query with join on `Help Article` and `Web Page View` over route as common parameter, sorted by page view count descending and limit to 10 or whatever number you want.

Reply:
ok noted